rust-toolchain.toml

```diff
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "1.73.0"
+channel = "1.76.0"
```

Comment:
As we discussed, it would be great to pin `clap` as well.

Reply:
Maybe we could bump rust in a separate PR to make sure the performance doesn't change from the build changes